operator/pkg/manifest/shared.go

```diff
@@ -490,9 +493,44 @@ func unmarshalAndValidateIOP(iopsYAML string, force, allowUnknownField bool, l c
 		l.LogAndError("Run the command with the --force flag if you want to ignore the validation error and proceed.")
 		return iop, fmt.Errorf(errs.Error())
 	}
+
 	return iop, nil
 }
 
+// resolveConflictSettings When user uses configuration file or --set flag to change value for parameters, in some
+// cases a parameter can cause conflict with parameters specified in profiles. For example, minAvailable is
+// specified in all profiles, user settings either from configuration file or --set flag will always overlay with
+// one of the profiles, thus minAvailable setting will be always present, however, this parameter is muturally
+// exclusive with maxUnavailable. When maxUnavailable is also present in IOP, that means user has specified
+// maxUnavailable value and prefers using maxUnavailable instead of using default minAvailable. In cases like this,
+// Istio must resolve the conflict, otherwise, the user request will fail. This method provides an entrypoint
+// to resolve similar conflicts. If there are other cases, this method should be the place to add the logic to
+// resolve conflict.
+func resolveConflictSettings(iop *iopv1alpha1.IstioOperator, l clog.Logger) {
+	if iop.Spec != nil && iop.Spec.Components != nil && iop.Spec.Components.Pilot != nil {
+		if iop.Spec.Components.Pilot.K8S == nil {
+			iop.Spec.Components.Pilot.K8S = &v1alpha1.KubernetesResourcesSpec{
+				PodDisruptionBudget: &v1alpha1.PodDisruptionBudgetSpec{MinAvailable: 1},
+			}
+		}
+		if iop.Spec.Components.Pilot.K8S.PodDisruptionBudget == nil {
+			iop.Spec.Components.Pilot.K8S.PodDisruptionBudget = &v1alpha1.PodDisruptionBudgetSpec{MinAvailable: 1}
+		}
+		if iop.Spec.Components.Pilot.K8S.PodDisruptionBudget.MaxUnavailable == 0 &&
```

Comment:
This will fail if the `MaxUnavailable` is provided in `%`, it can be either `int` or `percentage`. See [here](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#rounding-logic-when-specifying-percentages).

Reply:
@shamsher31 Thanks for reviewing the PR. I think that our current implementation for PodDisruptionBudget's MaxUnavailable and MinAvailable can only be unsigned int which excludes specifying the values as string such as 25%. If you run the following command, you will get an error.
Given this config.yaml file
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: example-istiocontrolplane
  namespace: istio-operator
spec:
  components:
    pilot:
        k8s:
          podDisruptionBudget:
            maxUnavailable: 25%
```

Then run this command
```
istioctl manifest generate -f config.yaml
```

You will get this error:
```
Error: invalid character '%' after top-level value:

YAML:
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: example-istiocontrolplane
  namespace: istio-operator
spec:
  components:
    pilot:
      k8s:
        podDisruptionBudget:
          maxUnavailable: 25%
```

Reply:
If you do not want to use config file, you can also try the following command which resolves same error.

```
istioctl manifest generate --set spec.components.pilot.k8s.podDisruptionBudget.minAvailable=25%
```
So the problem is that istioctl does not support setting these two to be a percentage even though k8s allows.

Reply:
@litong01 Ahh, Sorry I forget to mention, The API is updated recently and now it supports both cases. https://github.com/istio/api/pull/1952.

Reply:
@shamsher31 not an issue, will make changes accordingly based on the new PR. Will wait a bit for that PR to be in.

Reply:
@shamsher31 latest PR addressed the issue. Please take a look.